### ISSUE TITLE
Resistance 3: wrong water effects / rain drops

### ISSUE THREAD

**Issue by digant73** [RPCS3/rpcs3#15928]
### Quick summary

Just tested in deep the game playing it until the conclusion and noted crashes, graphical issues etc.

Target of this bug report is to fix the water effect issues reported below on the **GRAPHICAL GLITCHES:** section.
Sections **USED CUSTOM SETTINGS:** and **CRASHES DUE TO BUGS:** are provided just to help the developers on setup and to alert them on other bugs affecting the game (e.g. crash due to Lost-Device also present in other games) in case they want to spend time also on fixing them.

**USED CUSTOM SETTINGS:**
Despite the fact the R3's wiki reports to change (compared to default settings) only "ZCULL Accuracy" to "Relaxed", proper settings to use with current RPCS3 seems to be (tested on different HW):
- "ZCULL Accuracy" set to "Approximate" (of course Precise is usable but it impacts performance a little more):  ZCULL Accuracy set to Approximate makes performance a lot better than Relaxed (e.g. it also fixes the huge fps drop when zooming)
- "Write Color Buffer" set to "enabled": Write Color Buffer enabled fixes a graphical glitch present in the "Ambush" section in the "ST. Louis, MO" chapter

**CRASHES DUE TO BUGS:**
- The crashes (not so much frequent) are basically due to **Lost-Device** error reported on the log

**GRAPHICAL GLITCHES:**
- Water effects and water drops: They are present in different sections such as "Boat Ride" section and next ones in the "Haven, OK" chapter (water in the river is not rendered properly), "Ambush" section and next ones in "ST. Louis, MO" chapter (rain drops are not rendered properly on the ground) and "Forest" section and next ones in the "MT. Pleasant, PA" (water in the river is not rendered properly).

Attached a picture of the water issue on "Forest" section in the "MT. Pleasant, PA" chapter.

### Attach a log file

![water_on the_river](https://github.com/user-attachments/assets/8a691e51-5b75-4f5b-bf6d-0baf72fce5ee)

[RPCS3.log.gz](https://github.com/user-attachments/files/16590225/RPCS3.log.gz)

**Comment by Jonathan44062** [comment]
The Broken water texture is a known issue in all Insomniac Games (Ratchet & Clank, Resistance), see https://github.com/RPCS3/rpcs3/issues/12439

**Comment by digant73** [comment]
> The Broken water texture is a known issue in all Insomniac Games (Ratchet & Clank, Resistance), see #12439

thanks for the feedback. I see that the issue #12439 is open since 2022 and that @kd-11 auto-assigned the bug investigation. Hopefully he will have a look also to this bug. I have no idea if a fix is general for all games or it is specific for a single game.

**Comment by JimScript** [comment]
A thing to note about the regression is that it only affects Insomniac's deferred rendering titles(i.e. from Resistance 2 and beyond): Fall of Man flat out doesn't have this water shader issue, but Tools of Destruction and Quest for Booty both still have a water shader with problems identical to the rest of the series before the regression happened, so it's likely that the regression and the original issue are separate problems.

**Comment by digant73** [comment]
> A thing to note about the regression is that it only affects Insomniac's deferred rendering titles(i.e. from Resistance 2 and beyond): Fall of Man flat out doesn't have this water shader issue, but Tools of Destruction and Quest for Booty both still have a water shader with problems identical to the rest of the series before the regression happened, so it's likely that the regression and the original issue are separate problems.

yes I added a note on that on the bug report's summary. Before the regression the water effects were different than what I see on a PS3 but not so bad. For sure with the regression the effects are a lot worse and visibly broken

**Comment by kd-11** [comment]
I made some progress on this, but I need more hardware tests. At this point I might as well just stop debugging the regressions and start writing tests.

**Comment by digant73** [comment]
so good news. That fix will probably fix most of the same issues on other Insomniac games. Waiting for testing a PR or private branch for this issue and hopefully for KZ3

**Comment by kd-11** [comment]
Needs a retest with https://github.com/RPCS3/rpcs3/pull/18216

**Comment by alexconsole04** [comment]
It has improved, but is blocky. 
<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/962b1d0c-de9c-41f7-b6d0-c5c284b6523a" />

[RPCS3.log.gz](https://github.com/user-attachments/files/25459806/RPCS3.log.gz)

**Comment by digant73** [comment]
yes right, I provided all the videos comparing PS3 and PR #18216 directly in #18216
